### PR TITLE
Added tox and fixed all style errors

### PR DESCRIPTION
**Pull request by F3real**

### CODE REVIEW

**Comment by armenzg** [comment]
Hi @F3real I've quickly looked and one of the reasons that the PR is not showing well the differences is because the indentation has changed.
Could you please fix that?
It should be four spaces instead of eight.

**Comment by armenzg** [comment]
@chmanchester I'm happy to merge this PR. Anything to add?

This is in preparation to start using mozci and longer term to make the code easier to port into pulse_actions.

File: triggerbot/tree_watcher.py [hunk 1]
```diff
@@ -96,15 +94,12 @@ def _get_jobs(self, branch, rev, hidden):
...
 
-
     def get_hidden_jobs(self, branch, rev):
         return self._get_jobs(branch, rev, True)
 
-
     def get_visible_jobs(self, branch, rev):
         return self._get_jobs(branch, rev, False)
 
-
```
Comment (chmanchester) [inline triggerbot/tree_watcher.py]: I see newlines being removed here, bringing us down to one line between methods, but added in dump_buildstats.py, bringing that file up to two lines between methods. Please make them consistent (I don't have a preference for one or two at this point, I think they're mixed right now).

File: triggerbot/tree_watcher.py [hunk 2]
```diff
@@ -274,7 +263,7 @@ def attempt_triggers(self, branch, rev, builder, count, seen=0, attempt=0):
 
         self.log.info("Found %s jobs total for %s" % (rev_total, rev))
         if (seen * self.failure_tolerance_factor > rev_total and
-            seen > self.lower_trigger_limit):
+                seen > self.lower_trigger_limit):
```
Comment (chmanchester) [inline triggerbot/tree_watcher.py]: Please revert this change, I think the original indentation is preferable.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    coverage
+    flake8
+    pytest
+    mock
+    requests
+    treeherder-client
+
+commands =
+    flake8 triggerbot test buildapistats
+    coverage run --source=triggerbot -m py.test test
```
Comment (chmanchester) [inline tox.ini]: I haven't used pytest before, but I can't tell whether this is going to work as expected. The tests we have are based on the unittest api, will tox/pytest detect and run those? A good way to test this would be to run the coverage command and see what it reports (and possibly post the results here, I can tell you whether they seem accurate given the tests we have).

File: triggerbot/tree_watcher.py (hunk 1 above)
Comment (F3real) [inline triggerbot/tree_watcher.py]: It's default for pep8 standard, 2 newlines between functions and 1 newline between instance methods.

File: triggerbot/tree_watcher.py (hunk 2 above)
Comment (F3real) [inline triggerbot/tree_watcher.py]: Hmm, I can revert change but then we would fail at tox check.

File: triggerbot/tree_watcher.py (hunk 1 above)
Comment (chmanchester) [inline triggerbot/tree_watcher.py]: I did not know that! I guess this is correct then, thanks :)

File: triggerbot/tree_watcher.py (hunk 2 above)
Comment (chmanchester) [inline triggerbot/tree_watcher.py]: Hm, that's too bad. Feel free to leave it in then.

File: tox.ini (hunk 3 above)
Comment (F3real) [inline tox.ini]: `Name                             Stmts   Miss  Cover
----------------------------------------------------
triggerbot\__init__.py               0      0   100%
triggerbot\tree_watcher.py         202     80    60%
triggerbot\triggerbot_pulse.py     105    105     0%
----------------------------------------------------
TOTAL                              307    185    40%`

File: setup.py [hunk 4]
```diff
@@ -4,17 +4,22 @@
...
+      name='trigger-bot',
+      version='0.1',
+      packages=find_packages(),
+      entry_points={
+          'console_scripts': [
+              'run-trigger-bot = triggerbot.triggerbot_pulse:run',
+           ],
+      },
+      install_requires=[
+           'buildapi-client>=0.1'
```
Comment (chmanchester) [inline setup.py]: I don't see this package being used, is this for a later commit?

File: tox.ini (hunk 3 above)
Comment (chmanchester) [inline tox.ini]: Ok, that looks reasonable, it must be automatically detecting the tests. Thanks.

**Comment by chmanchester** [comment]
@armenzg you're ok to merge this at your convenience. Thanks for the patch @F3real !
